src/main/kotlin/com/fasterxml/jackson/module/kotlin/KotlinAnnotationIntrospector.kt

```diff
@@ -81,11 +87,15 @@ internal class KotlinAnnotationIntrospector(private val context: Module.SetupCon
 
     // Perform proper serialization even if the value wrapped by the value class is null.
     // If value is a non-null object type, it must not be reboxing.
-    override fun findNullSerializer(am: Annotated): JsonSerializer<*>? = (am as? AnnotatedMethod)?.let { _ ->
-        cache.findValueClassReturnType(am)
-            ?.takeIf { it.requireRebox() }
-            ?.let { cache.getValueClassBoxConverter(am.rawReturnType, it).delegatingSerializer }
-    }
+    override fun findNullSerializer(am: Annotated): JsonSerializer<*>? = (am as? AnnotatedMethod)
+        ?.ktClass()
+        ?.takeIf { it.requireRebox() }
+        ?.let { cache.getValueClassBoxConverter(am.rawReturnType, it).delegatingSerializer }
+
+    override fun findSerializer(am: Annotated): Any? = when ((am as? AnnotatedMethod)?.ktClass()) {
+        Duration::class -> KotlinToJavaDurationConverter.delegatingSerializer.takeIf { useJavaDurationConversion }
+        else -> null
+    } ?: super.findSerializer(am)
```

Comment:
This appears unnecessary since `findSerializationConverter` is implemented.
https://github.com/ProjectMapK/jackson-module-kogera/blob/2.15.2-beta1/src/main/kotlin/io/github/projectmapk/jackson/module/kogera/annotation_introspector/KotlinFallbackAnnotationIntrospector.kt

Reply:
This method is solution to nullable fields you mentioned [here](https://github.com/FasterXML/jackson-module-kotlin/pull/689#discussion_r1277908692):
```
// Need to define in nullable due to bug related to value class
@field:JsonFormat(shape = JsonFormat.Shape.STRING)
val v2: KotlinDuration? = 1.toDuration(DurationUnit.HOURS)
```

If I comment line 96 then `"should serialize Kotlin duration exactly as Java duration"` test will fail:
```
Expected :{"plain":3600.000000000,"optPlain":3600.000000000,"shapeAnnotation":"PT1H","optShapeAnnotation":"PT1H"}
Actual   :{"plain":3600.000000000,"optPlain":3600.000000000,"shapeAnnotation":3600.000000000,"optShapeAnnotation":"PT1H"}
```
Though I can remove `super` call fallback if you want.

Reply:
Okay, I guess I misunderstood the function.

However, let me make it a policy to use only `Converter` as a way of doing things.
An example is pushed below.
https://github.com/k163377/jackson-module-kotlin/compare/a0151d422267109a246c95c18242633b235c3ae7..05d3622b670a016f0daf253d50a914203e7ce586

The policy of using only a `Converter` will only subdivide within existing conditions, thus reducing the load compared to adding a `findSerializer`.

Reply:
Okay, cherry-picked this commit.